frontend/src/metabase/components/ErrorPages/ErrorDiagnosticModal.tsx

```diff
@@ -0,0 +1,239 @@
+import { capitalize } from "metabase/lib/formatting";
+import {
+  Button,
+  Center,
+  Icon,
+  Loader,
+  Text,
+  Stack,
+  Box,
+  Modal,
+  Flex,
+} from "metabase/ui";
+
+import type { ErrorPayload } from "./types";
+import { useErrorInfo } from "./use-error-info";
+import { downloadObjectAsJson, hasQueryData } from "./utils";
+
+interface ErrorDiagnosticModalProps {
+  errorInfo: ErrorPayload;
+  onClose: () => void;
+}
+
+type PayloadSelection = Partial<Record<keyof ErrorPayload, boolean>>;
+
+export const ErrorDiagnosticModal = ({
+  errorInfo,
+  onClose,
+}: ErrorDiagnosticModalProps) => {
+  const handleSubmit = (values: PayloadSelection) => {
+    const selectedKeys = Object.keys(values).filter(
+      key => values[key as keyof PayloadSelection],
+    );
+    const selectedInfo: ErrorPayload = _.pick(errorInfo, ...selectedKeys);
+
+    downloadObjectAsJson(
+      selectedInfo,
+      `metabase-diagnostic-info-${new Date().toISOString()}`,
+    );
+    onClose();
+  };
+
+  const canIncludeQueryData = hasQueryData(errorInfo?.entityName);
+
+  const hiddenValues = {
+    url: true,
+    entityName: true,
+  };
+
+  return (
+    <Modal
+      opened
+      onClose={onClose}
+      title={t`Download diagnostic information`}
+      padding="xl"
+      size="lg"
+    >
+      <FormProvider
+        initialValues={{
+          ...hiddenValues,
+          queryResults: false,
+          entityInfo: true,
+          frontendErrors: true,
+          backendErrors: true,
+          userLogs: true,
+          logs: true,
+          instanceInfo: true,
+        }}
+        onSubmit={handleSubmit}
+      >
+        <Form>
+          <Text>
+            {t`Select the info you want to include in the diagnostic JSON file.`}
+          </Text>
+          <Stack spacing="md" my="lg">
+            {canIncludeQueryData && (
+              <FormCheckbox name="queryResults" label={t`Query results`} />
+            )}
+            {!!errorInfo.localizedEntityName && (
+              <FormCheckbox
+                name="entityInfo"
+                label={t`${capitalize(
+                  errorInfo.localizedEntityName,
+                )} definition`}
+              />
+            )}
+            <FormCheckbox
+              name="frontendErrors"
+              label={t`Browser error messages`}
+            />
+            {!!errorInfo?.logs && (
+              <>
+                <FormCheckbox
+                  name="backendErrors"
+                  label={t`All server error messages`}
+                />
+                <FormCheckbox name="logs" label={t`All server logs`} />
+                <FormCheckbox
+                  name="userLogs"
+                  label={t`Server logs from the current user only`}
+                />
+              </>
+            )}
+            <FormCheckbox
+              name="instanceInfo"
+              // eslint-disable-next-line no-literal-metabase-strings -- we're mucking around in the software here
```

Comment:
Maybe
```
// eslint-disable-next-line no-literal-metabase-strings -- In this case 'Metabase' is OK
```

I don't want to poop on the party but I didn't understand the context of the "mucking around" comment

Reply:
As I understand it, the purpose of this rule is to be able to whitelabel the major user-facing strings to something like "Acme Co" - but it doesn't make sense to download "Acme Co" instance information, we're actually talking technically about the metabase software, not the data analytics platform.

Reply:
@luizarakaki do you agree that we should not label this string? `Metabase instance version and settings`